airbyte-integrations/connectors/destination-duckdb/destination_duckdb/destination.py

```diff
@@ -109,52 +110,40 @@ def write(
 
             con.execute(query)
 
-        buffer = defaultdict(list)
+        buffer = defaultdict(lambda: defaultdict(list))
 
         for message in input_messages:
             if message.type == Type.STATE:
                 # flush the buffer
                 for stream_name in buffer.keys():
                     logger.info(f"flushing buffer for state: {message}")
                     table_name = f"_airbyte_raw_{stream_name}"
-                    query = f"""
-                    INSERT INTO {schema_name}.{table_name}
-                      (_airbyte_ab_id, _airbyte_emitted_at, _airbyte_data)
-                    VALUES (?,?,?)
-                    """
-                    con.executemany(query, buffer[stream_name])
+                    pa_table = pa.Table.from_pydict(buffer[stream_name])
```

Comment:
@hrl20 - Do you know how this method will deal with nested dictionaries in the `airbyte_raw` column? In my experience, there have been a few cases where PyArrow seems to struggle on oddly shaped schemas...

1. Nested nodes that have no children, basically an empty dictionary: `subfield_a: {}`
2. Challenges distinguishing child nodes as "dictionaries" vs "structs", and the implications and requirements of each.
3. Challenges dealing with subsequent records which conflict with each other's inferred typing: for instance `{'subfield_a': True}` followed by `{'subfield_a': 'True'}` or `{'subfield_a': 42}`.

I'm curious if you've run into these, and if you have any thoughts on them.

Reply:
hmm, let me see if I can produce some data that forces errors, and maybe make the connector fall back to the old behavior if such exception arises?

Reply:
I love that idea! A little bit messier perhaps, but I think (if needed) it's a nice balance between prioritizing speed and "don't fall down" . 😅

Reply:
I tried some of these cases but none of them triggered a PyArrow error, so at this point I don't know what exception might be raised. I still added the error handling/fallback way and confirmed they worked when I forced exceptions to be thrown.